utils/staking-miner/src/opts.rs

```diff
@@ -0,0 +1,275 @@
+///
+/// Possible options:
+/// * --submission-strategy if-leading: only submit if leading
+/// * --submission-strategy always: always submit
+/// * --submission-strategy "percent-better <percent>": submit if submission is `n` percent better.
+///
+impl FromStr for SubmissionStrategy {
+	type Err = String;
+
+	fn from_str(s: &str) -> Result<Self, Self::Err> {
+		let s = s.trim();
+
+		let res = if s == "if-leading" {
+			Self::IfLeading
+		} else if s == "always" {
+			Self::Always
+		} else if s.starts_with("percent-better ") {
+			let percent: u32 = s[15..].parse().map_err(|e| format!("{:?}", e))?;
+			Self::ClaimBetterThan(Perbill::from_percent(percent))
+		} else {
+			return Err(s.into())
+		};
+		Ok(res)
+	}
+}
+
+#[cfg(test)]
+mod test_super {
+	use super::*;
+
+	#[test]
+	fn cli_monitor_works() {
+		let opt = Opt::try_parse_from([
+			env!("CARGO_PKG_NAME"),
+			"--uri",
+			"hi",
+			"monitor",
+			"--seed-or-path",
+			"//Alice",
+			"--listen",
+			"head",
+			"seq-phragmen",
+		])
+		.unwrap();
+
+		assert_eq!(
+			opt,
+			Opt {
+				uri: "hi".to_string(),
+				command: Command::Monitor(MonitorConfig {
+					seed_or_path: "//Alice".to_string(),
+					listen: "head".to_string(),
+					solver: Solver::SeqPhragmen { iterations: 10 },
+					submission_strategy: SubmissionStrategy::IfLeading,
+				}),
+			}
+		);
+	}
+
+	#[test]
+	fn cli_dry_run_works() {
+		let opt = Opt::try_parse_from([
+			env!("CARGO_PKG_NAME"),
+			"--uri",
+			"hi",
+			"dry-run",
+			"--seed-or-path",
+			"//Alice",
+			"phrag-mms",
+		])
+		.unwrap();
+
+		assert_eq!(
+			opt,
+			Opt {
+				uri: "hi".to_string(),
+				command: Command::DryRun(DryRunConfig {
+					seed_or_path: "//Alice".to_string(),
+					at: None,
+					solver: Solver::PhragMMS { iterations: 10 },
+					force_snapshot: false,
+				}),
+			}
+		);
+	}
+
+	#[test]
+	fn cli_emergency_works() {
+		let opt = Opt::try_parse_from([
+			env!("CARGO_PKG_NAME"),
+			"--uri",
+			"hi",
+			"emergency-solution",
+			"99",
+			"phrag-mms",
+			"--iterations",
+			"1337",
+		])
+		.unwrap();
+
+		assert_eq!(
+			opt,
+			Opt {
+				uri: "hi".to_string(),
+				command: Command::EmergencySolution(EmergencySolutionConfig {
+					take: Some(99),
+					at: None,
+					solver: Solver::PhragMMS { iterations: 1337 }
+				}),
+			}
+		);
+	}
+
+	#[test]
+	fn cli_info_works() {
+		let opt = Opt::try_parse_from([env!("CARGO_PKG_NAME"), "--uri", "hi", "info"]).unwrap();
+
+		assert_eq!(opt, Opt { uri: "hi".to_string(), command: Command::Info(InfoOpts {}) });
+	}
+
+	#[test]
+	fn submission_strategy_from_str_works() {
+		use std::str::FromStr;
+
+		assert_eq!(SubmissionStrategy::from_str("if-leading"), Ok(SubmissionStrategy::IfLeading));
+		assert_eq!(SubmissionStrategy::from_str("always"), Ok(SubmissionStrategy::Always));
+		assert_eq!(
+			SubmissionStrategy::from_str("  percent-better 99   "),
+			Ok(SubmissionStrategy::ClaimBetterThan(Perbill::from_percent(99)))
+		);
+	}
+}
```

Comment:
I just noticed that there are no integration tests.  
Do you want to add one? Otherwise I would volunteer

Reply:
Done, thanks